firebase-firestore/src/main/java/com/google/firebase/firestore/model/mutation/TransformMutation.java

```diff
@@ -123,6 +123,10 @@ public MaybeDocument applyToLocalView(
 
   @Override
   public FieldMask getFieldMask() {
+    // Note: Theoretically, we should only include non-idempotent fields in this field mask as this
+    // mask is used to populate the base state for all DocumentTransforms.  By including  all
+    // fields, we incorrectly prevent rebasing of idempotent transforms (such as `arrayUnion()`)
+    // when any non-idempotent transforms are present.
```

Comment:
This is a good comment, but it may not make sense out-of-context (it assumes knowledge of how this method is going to be used).  I'd actually move it into LocalStore where getFieldMask() is called rather than having it here.

Reply:
Yeah, that makes sense. It's basically the same reason as to why I don't want to exclude the idempotent fields here.